internal/controllers/synthesis/connection.go

```diff
@@ -71,7 +71,12 @@ func (s *SynthesizerPodConnection) Synthesize(ctx context.Context, syn *apiv1.Sy
 		return nil, fmt.Errorf("creating remote command executor: %w", err)
 	}
 
-	streamCtx, cancel := context.WithTimeout(ctx, syn.Spec.ExecTimeout.Duration)
+	// Execs will in most cases block pod termination, so use the min of the exec and pod lifespan timeouts
+	timeout := syn.Spec.ExecTimeout.Duration
+	if pt := syn.Spec.PodTimeout.Duration; pt < timeout {
```

Comment:
Can this be somehow specified with kubebuilder and/or [CEL expressions](https://kubernetes.io/docs/reference/using-api/cel/)?